unison-share-api/src/Unison/Server/Doc.hs

```diff
@@ -303,29 +416,85 @@ renderDoc pped terms typeOf eval types tm =
                 acc' = case tm of
                   Term.Ref' r
                     | Set.notMember r seen ->
-                      (: acc) . Term . (Reference.toText r,) <$> case r of
```

Comment:
There are some strange choices being made in this block, I just persisted the existing behaviour.

E.g. if we encounter a builtin reference, we render its type signature into a `BuiltinObject` 🤔 
I'm sure the result is good, but the implementation seemed a bit strange so I tried to clarify a bit by using the `EvaluatedSrcTerm` type.